spec/datadog/tracing/contrib/elasticsearch/transport_spec.rb

```diff
@@ -175,9 +175,17 @@ def call(env)
     end
   end
 
-  describe 'client configuration override' do
+  describe 'transport configuration override' do
     context 'when #service is overridden' do
-      before { Datadog.configure_onto(client.transport, service_name: service_name) }
+      let(:version_greater_than_8) { Gem::Version.new(::Elasticsearch::VERSION) >= Gem::Version.new('8.0.0') }
+
+      before do
+        if version_greater_than_8
+          Datadog.configure_onto(client.transport, service_name: service_name)
+        else
+          Datadog.configure_onto(client.transport.transport, service_name: service_name)
```

Comment:
@marcotc @TonyCTHsu When I added back this test to allow custom config on the transport object it initially failed. Turns out that prior to ES 8.0 `client.transport` refers to the client instance. `client.transport.transport` refers to the transport instance. With ES > 8.0 `client.transport` refers to the transport instance. So when I changed the patch of `perform_request` to not check `Datadog.configuration_for(self, :service_name)` and only do `Datadog.configuration_for(transport, :service_name)` it was getting back nil on ES 7.0 since there is a discrepancy between what we set with `configure_onto` and what we get with `configuration_for`.  

As I see it we have 3 options:
1. Keep my changes as they are, and update the docs to explain that if you are on ES < 8.0 you need to configure_onto client.transport.transport. This makes our code simpler, but our docs slightly more complicated.
2. Keep my code changes, update the this test to skip ES < 8.0, and update our documentation to say we don't support custom configuration for transport below ES 8.0
3. Change the code so that if ES < 8 we support configuration on the client object, but not support it for 8.0+ and remove the deprecation warning.

I think option #1 is the best balance of user-friendliness and code simplicity.